src/nplinker/scoring/link_graph.py

```diff
@@ -267,6 +258,101 @@ def filter(self, u_nodes: Sequence[Entity], v_nodes: Sequence[Entity] = [], /) -
 
         return lg
 
+    @validate_uv
+    def get_link_data(
+        self,
+        u: Entity,
+        v: Entity,
+    ) -> LINK_DATA | None:
+        """Get the data for a link between two objects.
+
+        Args:
+            u: the first object, either a GCF, Spectrum, or MolecularFamily
+            v: the second object, either a GCF, Spectrum, or MolecularFamily
+
+        Returns:
+            A dictionary of scoring methods and their data for the link between the two objects, or
+            None if there is no link between the two objects.
+
+        Examples:
+            >>> lg.get_link_data(gcf, spectrum)
+            {"metcalf": Score("metcalf", 1.0, {"cutoff": 0.5})}
+        """
+        return self._g.get_edge_data(u, v)  # type: ignore
+
+    def get_table_data(self, display_limit: int | None = None) -> list[dict[str, Any]]:
```

Comment:
It seems this method can be a private method. 

Also, I prefer renaming it to e.g. `_links_to_dicts`.